flask/user.py

```diff
@@ -0,0 +1,111 @@
+from app import app
+from mongo import db
+from flask import render_template
+from flask import request, make_response, jsonify
+from datetime import datetime
+from auth import generate_uuid
+
+@app.route("/users/<username>/", methods=["GET"])
+def profile(username):
+    uuid = request.cookies.get("uuid")
+    if not uuid:
+        return render_template("login.html")
+
+    user = db.users.find_one({"uuid": uuid})
+
+    if not user or user["name"] != username:
+        return render_template("login.html")
+
+    # return his owned and maintained packages
+    packages = db.packages.find(
+        {"$or": [{"author": user["_id"]}, {"maintainers": user["_id"]}]}
+    )
+
+    return render_template(
+        "profile.html", user=user, packages=[package for package in packages]
+    )
+
+@app.route('/packages', methods=["GET"])
+def search_packages():
+    query = request.args.get('query')
+    packages = db.packages.find(
+        {
+            "$or": [
+                {"name": {"$regex": query}},
+                {"tags": {"$in": [query]}},
+                {"description": {"$regex": query}},
+            ]
+        }
+    )
+
+    return jsonify([package for package in packages])
+
+@app.route("/packages", methods=["POST"])
```

Comment:
How about creating a new file like ``package.py`` for the routes related to the package.

Reply:
Sure, I had created for auth, mongo, maybe I missed that.

Reply:
Cool ! I am also rebasing my PR on top of this one in order to keep it updated.